akka-stream-tests/src/test/scala/akka/stream/io/TlsSpec.scala

```diff
@@ -113,11 +114,39 @@ class TlsSpec extends StreamSpec(TlsSpec.configOverrides) with WithLogCapturing
       x
     }
 
-    val cipherSuites =
-      NegotiateNewSession.withCipherSuites("TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA", "TLS_RSA_WITH_AES_128_CBC_SHA")
-    def clientTls(closing: TLSClosing) = TLS(sslContext, None, cipherSuites, Client, closing)
-    def badClientTls(closing: TLSClosing) = TLS(initWithTrust("/badtruststore"), None, cipherSuites, Client, closing)
-    def serverTls(closing: TLSClosing) = TLS(sslContext, None, cipherSuites, Server, closing)
+    def createSSLEngine(context: SSLContext, role: TLSRole): SSLEngine =
+      createSSLEngine2(context, role, hostnameVerification = false, hostInfo = None)
+
+    def createSSLEngine2(
+        context: SSLContext,
+        role: TLSRole,
+        hostnameVerification: Boolean,
+        hostInfo: Option[(String, Int)]): SSLEngine = {
+
+      val engine = hostInfo match {
+        case None                   => context.createSSLEngine()
+        case Some((hostname, port)) => context.createSSLEngine(hostname, port)
+      }
+
+      if (hostnameVerification && role == akka.stream.Client) {
```

Comment:
If `hostnameVerification == true && hostInfo.isEmpty`, hostname verification will still not work because the SSLEngine doesn't know about the hostname. Might make sense to add an assertion for that.

Reply:
:+1: since it makes the test more explicit